File: infra/src/main/java/com/beat/infra/persistence/user/mapper/UsersPersistenceMapper.java
package com.beat.infra.persistence.user.mapper;

import org.springframework.stereotype.Component;

import com.beat.domain.user.domain.Users;
import com.beat.infra.persistence.user.entity.UsersJpaEntity;

@Component
public class UsersPersistenceMapper {

	public Users toDomain(UsersJpaEntity entity) {
		return Users.rehydrate(
			entity.getId(),
			entity.getRole()
		);
	}

	public UsersJpaEntity toEntity(Users domain) {
		return UsersJpaEntity.rehydrate(
			domain.getId(),
			domain.getRole()
		);
	}
}

File: infra/src/main/kotlin/com/beat/infra/persistence/user/mapper/UsersPersistenceMapper.kt
package com.beat.infra.persistence.user.mapper

import com.beat.domain.user.model.Users
import com.beat.infra.persistence.user.entity.UsersJpaEntity
import org.springframework.stereotype.Component

@Component
class UsersPersistenceMapper {
    fun toDomain(entity: UsersJpaEntity): Users =
        Users.rehydrate(entity.id, entity.role)

    fun toEntity(domain: Users): UsersJpaEntity =
        UsersJpaEntity.rehydrate(domain.getId(), domain.role)
}
